File: frontend/vue-vben-admin/apps/web-antd/src/api/system/menu.ts
import { requestClient } from '#/api/request';

export interface MenuTreeNode {
  children: MenuTreeNode[];
  id: string;
  name: string;
  title: string;
}

export interface MenuManagementItem {
  affixTab: boolean;
  children: MenuManagementItem[];
  component?: null | string;
  icon?: null | string;
  id: string;
  isEnabled: boolean;
  isVisible: boolean;
  name: string;
  order: number;
  parentId?: null | string;
  path: string;
  permissionCode?: null | string;
  redirect?: null | string;
  title: string;
}

export interface SaveMenuParams {
  affixTab: boolean;
  component?: null | string;
  icon?: null | string;
  isEnabled: boolean;
  isVisible: boolean;
  name: string;
  order: number;
  parentId?: null | string;
  path: string;
  permissionCode?: null | string;
  redirect?: null | string;
  title: string;
}

export async function getMenuTreeApi() {
  return requestClient.get<MenuTreeNode[]>('/system/menu/tree');
}

export async function getMenuListApi() {
  return requestClient.get<MenuManagementItem[]>('/system/menu/list');
}

export async function createMenuApi(data: SaveMenuParams) {
  return requestClient.post<MenuManagementItem>('/system/menu', data);
}

export async function updateMenuApi(id: string, data: SaveMenuParams) {
  return requestClient.put<MenuManagementItem>(`/system/menu/${id}`, data);
}

export async function deleteMenuApi(id: string) {
  return requestClient.delete<boolean>(`/system/menu/${id}`);
}

File: src/MiniAdmin.Domain/Entities/Menu.cs
namespace MiniAdmin.Domain.Entities;

public sealed class Menu
{
    public Guid Id { get; set; }

    public Guid? ParentId { get; set; }

    public Menu? Parent { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Path { get; set; } = string.Empty;

    public string? Component { get; set; }

    public string? Redirect { get; set; }

    public string Title { get; set; } = string.Empty;

    public string? Icon { get; set; }

    public int Order { get; set; }

    public bool AffixTab { get; set; }

    public string? PermissionCode { get; set; }

    public bool IsEnabled { get; set; } = true;

    public bool IsVisible { get; set; } = true;

    public List<Menu> Children { get; set; } = [];

    public List<RoleMenu> RoleMenus { get; set; } = [];
}
